File: Olympiad/COCI/official/2007/contest5_solutions/trik.c

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 5 - Task TRIK
  Programming language C
*/

#include <stdio.h>

int main()
{
   char s[51];
   int gdje = 1, i;
   scanf( "%s", s );

   for ( i=0; s[i]!='\0'; ++i ) {
      if      ( s[i] == 'A' && gdje != 3 ) gdje = 3-gdje;
      else if ( s[i] == 'B' && gdje != 1 ) gdje = 5-gdje;
      else if ( s[i] == 'C' && gdje != 2 ) gdje = 4-gdje;
   }

   printf( "%d\n", gdje );

   return 0;
}

File: Olympiad/COCI/official/2007/contest5_solutions/trik.pas

(*
  Croatian Open Competition in Informatics 2006/2007
  Contest 5 - Task TRIK
  Programming language Pascal
*)

program trik;

var
   s       : string;
   gdje, i : integer;

begin
   readln(s);

   gdje := 1;
   for i:=1 to length(s) do begin
      if (s[i]='A') and (gdje <> 3) then gdje := 3-gdje;
      if (s[i]='B') and (gdje <> 1) then gdje := 5-gdje;
      if (s[i]='C') and (gdje <> 2) then gdje := 4-gdje;
   end;

   writeln(gdje);
end.
